pkg/router/template/router_test.go

```diff
@@ -161,6 +193,23 @@ func TestDeleteEndpoints(t *testing.T) {
 		t.Errorf("Unable to find serivce unit %s after creation", suKey)
 	}
 
+	// Deleting endpoints without an associated route won't result
+	// in a state change so create an associated route for the
+	// service unit.
+	router.AddRoute(&routev1.Route{
```

Comment:
Could you add a similar route addition to `TestDeleteServiceUnit()` so your changes in this PR to `DeleteServiceUnit()` are covered by unit tests?